src/plugins/github/fetchGithubRepo.js

```diff
@@ -76,6 +79,23 @@ export default async function fetchGithubRepo(
   return ((mirror.extract(resolvedId): any): Repository);
 }
 
+// GitHub object IDs are urlsafe-base64-encoded strings that decode to
+// ASCII strings of the form "123:Typename4567[...]", where the "123"
+// numbers are a function only of the typename and the "4567" numbers
+// are the object's database ID, and the "[...]" is either empty or a
+// further section like ":commithash" for commits.
+//
+// See tests for `_guessTypename` for some example object IDs.
+const GITHUB_ID_TYPENAME_PATTERN = /^[0-9]*:([A-Za-z0-9_-]*[A-Za-z_-])[0-9]+(?:[^A-Za-z0-9_-].*)?$/;
```

Comment:
Alt: can use `/i` regex flag to be case-insensitive and shorten the pattern. Since we're not using any uppercase or lowercase specific matching.